frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/GpuAllocationManager.java

```diff
@@ -73,14 +77,28 @@ public synchronized Node allocateCandidateRequestNode(ResourceDescriptor request
 
       if (nodeLabel != null) {
         Set<String> nodeLabels = entry.getValue().getNodeLabels();
-        if (!nodeLabels.contains(nodeLabel)) {
-          LOGGER.logInfo(
+        if (nodeLabels != null && nodeLabels.size() > 0 && !nodeLabels.contains(nodeLabel)) {
+          LOGGER.logDebug(
               "allocateCandidateRequestNode: Skip node %s, label does not match:%s",
               entry.getValue().getHostName(), nodeLabel);
```

Comment:
log nodeLabels

Reply:
bugs

---
In reply to: [158486868](https://github.com/Microsoft/pai/pull/128#discussion_r158486868) [](ancestors = 158486868)